### PR TITLE
Adds more competitor pages

### PR DESCRIPTION
**Pull request by jay-khatri**
## Summary
- adds a datadog comparison page
- adds the comparison page to the footer of our home page (supposed to be good for SEO)

## How did you test this change?
- clicktest on the footer of the home page
- clicktest on ~/compare/highlight-vs-datadog (grammar check appreciated)

## Are there any deployment considerations?
no

### CODE REVIEW

File: docs-content/general/4_company/our-competitors.md [hunk 1]
```diff
@@ -5,10 +5,12 @@ createdAt: 2021-09-14T02:03:51.000Z
...
 ---
 
-We respect our competitors. In fact, we're not in the business of trying to convince people to use our product when it's not the right fit for them. Explore details about some of our competitors below:
+We respect our competitors. In fact, we're not in the business of trying to convince people to use our product when it's not the right fit for them. Explore details about some of our competitors below.
+
+## Session Replay Competitors
 
 <DocsCardGroup>
-    <DocsCard title="Log Rocket" href="https://highlight.io/compare/highlight-vs-logrocket">
+    <DocsCard title="Logrocket" href="https://highlight.io/compare/highlight-vs-logrocket">
```
Comment (Vadman97) [inline docs-content/general/4_company/our-competitors.md:13]:
```suggestion
    <DocsCard title="LogRocket" href="https://highlight.io/compare/highlight-vs-logrocket">
```

File: highlight.io/components/Competitors/datadog.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+	paragraphs: [
+		HighlightIsOpenAndTransparent('Datadog'),
+		HighlightShipsNewFeatures,
+		{
+			header: 'Full-stack Observability',
+			body: 'While Datadog provides a comprehensive set of separate features for observability, it does not support full-stack observability. [Highlight.io](https://highlight.io) enables teams to monitor and optimize their entire tech stack, pariging server-side infrastructure with your frontend web application. This makes Highlight.io a more comprehensive solution for developers who need to monitor their entire tech stack, in a simple, easy to implement solution.',
+		},
+		{
+			header: 'Pricing',
+			body: `Datadog's pricing structure is based on the number of hosts you have, the amount of data you ingest, and the amount of data you retain. This can make it difficult to predict your monthly bill. Highlight.io's pricing is strictly based on retention + usage and you find more details on our [pricing page](https://highlight.io/pricing).
```
Comment (Vadman97) [inline highlight.io/components/Competitors/datadog.ts:159]: > Highlight.io's pricing is strictly based on retention + usage

i would simplify this as
```Highlight.io's pricing is based entirely on your usage with a straightforward retention multiplier, and you can find more details on our [pricing page](https://highlight.io/pricing).```

**Review by Vadman97 — APPROVED** [review]

File: highlight.io/components/Competitors/datadog.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+				},
+			],
+		},
+	],
+	paragraphs: [
+		HighlightIsOpenAndTransparent('Datadog'),
+		HighlightShipsNewFeatures,
+		{
+			header: 'Full-stack Observability',
+			body: 'While Datadog provides a comprehensive set of separate features for observability, it does not support full-stack observability. [Highlight.io](https://highlight.io) enables teams to monitor and optimize their entire tech stack, pariging server-side infrastructure with your frontend web application. This makes Highlight.io a more comprehensive solution for developers who need to monitor their entire tech stack, in a simple, easy to implement solution.',
```
Comment (et) [inline highlight.io/components/Competitors/datadog.ts:155]: pariging?

File: highlight.io/components/Competitors/datadog.ts [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+		},
+		{
+			header: 'Pricing',
+			body: `Datadog's pricing structure is based on the number of hosts you have, the amount of data you ingest, and the amount of data you retain. This can make it difficult to predict your monthly bill. Highlight.io's pricing is strictly based on retention + usage and you find more details on our [pricing page](https://highlight.io/pricing).
+            `,
+		},
+		{
+			header: 'Agent-less Architecture',
+			body: `Datadog's architecture requires you to install an agent on your server. This can be a pain to install and maintain. Highlight.io's architecture is agent-less and requires no installation, but in the case that you want to install an agent long-term, we have an open-source agent/collector that you can install on your servers.`,
```
Comment (et) [inline highlight.io/components/Competitors/datadog.ts:164]: We have an agent?

Comment (jay-khatri) [reply highlight.io/components/Competitors/datadog.ts:164]: We have an OTEL gateway. We don't market it (yet).

File: highlight.io/components/Competitors/datadog.ts (hunk 2 above)
Comment (jay-khatri) [reply highlight.io/components/Competitors/datadog.ts:159]: sounds good.

**Pull request merged**